packages/realm_dart/lib/src/native/collection_changes_handle.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 MongoDB, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:ffi';
+
+import 'ffi.dart';
+
+import '../collections.dart';
+import 'from_native.dart';
+import 'handle_base.dart';
+import 'realm_bindings.dart';
+import 'realm_library.dart';
+
+class CollectionChangesHandle extends HandleBase<realm_collection_changes> {
+  CollectionChangesHandle(Pointer<realm_collection_changes> pointer) : super(pointer, 256);
+
+  CollectionChanges get changes {
+    return using((arena) {
+      final outNumDeletions = arena<Size>();
+      final outNumInsertions = arena<Size>();
+      final outNumModifications = arena<Size>();
+      final outNumMoves = arena<Size>();
+      final outCollectionCleared = arena<Bool>();
+      final outCollectionWasDeleted = arena<Bool>();
+      realmLib.realm_collection_changes_get_num_changes(
+        pointer,
+        outNumDeletions,
+        outNumInsertions,
+        outNumModifications,
+        outNumMoves,
+        outCollectionCleared,
+        outCollectionWasDeleted,
+      );
+
+      final deletionsCount = outNumDeletions != nullptr ? outNumDeletions.value : 0;
+      final insertionCount = outNumInsertions != nullptr ? outNumInsertions.value : 0;
+      final modificationCount = outNumModifications != nullptr ? outNumModifications.value : 0;
+      var moveCount = outNumMoves != nullptr ? outNumMoves.value : 0;
```

Comment:
26 vs 16 .. Maybe we should standardize on this, but I'll leave that to another day.

Reply:
I tend to prefer checking for `!=` because you have the check and the usage closer together. It's also similar to the null propagation operator for stuff that doesn't support it - e.g. you'd write that as `outNumMoves?.value ?? 0` if `nullptr` was supported by `?.`.

Reply:
Interesting take 🙂 Regarding the comparison to `outNumMoves?.value ?? 0`, agree that it's more similar, but I guess it's also a matter of preference for including or excluding explicit `!`. When the comparison is not against `null` or `nullptr`, the readability to me is better w/o `!`, and so it could be a case for consistency across the board 🤷‍♀️ But just a nit either way I think 😛